spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -514,44 +516,34 @@ private void handleAcks(Messages<T> messages) {
 		}
 
 		private void handleAck(Message<T> message) {
-			try {
-				this.consumer.acknowledge(message);
-			}
-			catch (PulsarClientException pce) {
-				this.consumer.negativeAcknowledge(message);
-			}
+			AbstractAcknowledgement.handleAckByMessageId(this.consumer, message.getMessageId());
 		}
 
 	}
 
-	private static final class ConsumerAcknowledgment implements Acknowledgement {
+	private static abstract class AbstractAcknowledgement implements Acknowledgement {
 
-		private final Consumer<?> consumer;
+		private static final LogAccessor logger = new LogAccessor(LogFactory.getLog(AbstractAcknowledgement.class));
 
-		private final Message<?> message;
+		protected final Consumer<?> consumer;
 
-		ConsumerAcknowledgment(Consumer<?> consumer, Message<?> message) {
+		AbstractAcknowledgement(Consumer<?> consumer) {
 			this.consumer = consumer;
-			this.message = message;
 		}
 
 		@Override
-		public void acknowledge() {
-			try {
-				this.consumer.acknowledge(this.message);
-			}
-			catch (PulsarClientException e) {
-				this.consumer.negativeAcknowledge(this.message);
-			}
+		public void acknowledge(MessageId messageId) {
+			handleAckByMessageId(this.consumer, messageId);
 		}
 
-		@Override
-		public void acknowledge(MessageId messageId) {
+		private static void handleAckByMessageId(Consumer<?> consumer, MessageId messageId) {
 			try {
-				this.consumer.acknowledge(messageId);
+				consumer.acknowledge(messageId);
 			}
-			catch (PulsarClientException e) {
-				this.consumer.negativeAcknowledge(messageId);
+			catch (PulsarClientException pce) {
+				AbstractAcknowledgement.logger.warn(pce,
```

Comment:
**[NIT]** Inconsistent qualifying as we use `AbstractAcknowledgement.logger` here but above on 536 we just call `handleAckByMessageId`. I am ok w/ or w/o the classname qualfier, as long as they are consistent

Reply:
Going to handle in subsequent PR.